### PR TITLE
Bundle and deploy

### PR DESCRIPTION
**Pull request by victorhurdugaci**
Few changes for testing on Nano server:
- Enable the MusicStore tests to run on CoreCLR
- Script that bundles the application, copies it to the remote machine, sets up a remote PS session and then runs the tests

Please review @Praburaj , @pranavkm

### CODE REVIEW

**Comment by victorhurdugaci** [comment]
@moozzyk you might be interested in this if you try to test something

File: test/E2ETests/project.json [hunk 1]
```diff
@@ -21,6 +21,14 @@
                 "System.Net.Http": "",
                 "System.Xml": ""
             }
+        },
+        "dnxcore50": {
```
Comment (Praburaj) [inline test/E2ETests/project.json]: Is there a way to disable running the coreclr version of the library on teamcity? I think the build.cmd runs once against all the targets configured.

Comment (Praburaj) [inline test/E2ETests/project.json]: running it on coreclr is a bit redundant on teamcity and its going to take more time.

File: test/E2ETests/project.json [hunk 2]
```diff
@@ -21,6 +21,14 @@
...
+            "dependencies": {
+                "Microsoft.Win32.Primitives": "4.0.0-beta-*",
```
Comment (Praburaj) [inline test/E2ETests/project.json]: Win32 Primitives might not be required anymore.

File: tools/BundleAndDeploy.ps1 [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    [string] $serverFolder,
+    [string] $userName,
+    [string] $password
+)
+
+function UpdateHostInProjectJson($projectFile, $newhost)
+{
+    (Get-Content $projectFile) | 
+    Foreach-Object {
+        $_ -replace "http://localhost", "http://$newhost"
```
Comment (Praburaj) [inline tools/BundleAndDeploy.ps1]: How do you set this appbase url in the test?

**Comment by moozzyk** [comment]
@victorhurdugaci looks like I was able to successfully bundle and deploy to nano server with this script. 

running MusicStore on nano failed but I don't think it is because of this script:
```
[10.135.33.3]: PS C:\dev\MusicStore> .\run
.\run : System.DllNotFoundException: Unable to load DLL 'sni.dll': The specified module could not be found. (Exception from HRESULT: 0x8007007E)
    + CategoryInfo          : NotSpecified: (System.DllNotFo...LT: 0x8007007E):String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError

   at System.Data.SqlClient.SNINativeMethodWrapper.UnmanagedIsTokenRestricted(IntPtr token, Boolean& isRestricted)

 at System.Data.Win32NativeMethods.IsTokenRestrictedWrapper(IntPtr token)
   at System.Data.ProviderBase.DbConnectionPoolIdentity.GetCurrent()
```

**Comment by victorhurdugaci** [comment]
Oh yes, the SQL client doesn't work. You have to use the in memory store

File: tools/BundleAndDeploy.ps1 [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+Write-Host "Project: $projectName"
+Write-Host "Work directory: $workDir"
+Write-Host "Remote dir: $remoteDir"
+Write-Host "User name: $userName"
+
+try
+{
+    if ($userName) {
+		net use $remoteRoot $password /USER:$userName
```
Comment (moozzyk) [inline tools/BundleAndDeploy.ps1]: nit: spaces

**Comment by Praburaj** [comment]
And how do you choose the app to use inmemory store on nano server?

**Comment by victorhurdugaci** [comment]
For now you edit [this line](https://github.com/aspnet/MusicStore/blob/master/src/MusicStore/Startup.cs#L33) :blush:

**Comment by Praburaj** [comment]
Hmm :smile: Probably you could have a config override like this. We already do this for IIS scenarios. https://github.com/aspnet/MusicStore/blob/dev/test/E2ETests/compiler/shared/Mocks/StartupSocialTesting.cs#L35

**Comment by victorhurdugaci** [comment]
It should use the new `IRuntimeEnvironment` to figure out if needs the in memory store

**Comment by Praburaj** [comment]
oh really. Does that give the host information as well?

**Comment by victorhurdugaci** [comment]
Yes, it gives:
- OS name
- OS version
- host version
- host type (clr/mono/coreclr)
- host architecture

**Comment by Praburaj** [comment]
ah.. I mean does that say if it is IIS / WebListener / Kestrel as well?

**Comment by victorhurdugaci** [comment]
oh, sorry. no, it only gives information about OS and DNX

File: test/E2ETests/project.json (hunk 1 above)
Comment (victorhurdugaci) [inline test/E2ETests/project.json]: Would be okay to run these tests only on CoreCLR? They spawn a server anyway, right?

**Pull request closed without merging**